multiversx_sdk_core/transaction_factories/delegation_factory.py

```diff
@@ -82,398 +66,266 @@ def add_nodes(self,
         ]
 
-        transaction = self.create_transaction(
+        transaction = self._create_transaction_intent(
             sender=sender,
             receiver=delegation_contract,
             data_parts=parts,
-            execution_gas_limit=self.config.gas_limit_delegation_operations + self.config.additional_gas_for_delegation_operations,
-            gas_limit_hint=gas_limit,
-            gas_price=gas_price,
-            nonce=transaction_nonce,
-            value=value,
-            guardian=guardian
+            execution_gas_limit=self.config.gas_limit_delegation_operations + self.config.additional_gas_for_delegation_operations
         )
 
         return transaction
 
-    def _compute_gas_limit(self, payload: ITransactionPayload, execution_gas: IGasLimit, has_guardian: bool) -> IGasLimit:
+    def _compute_gas_limit(self, payload: ITransactionPayload, execution_gas: int) -> int:
         data_movement_gas = self.config.min_gas_limit + self.config.gas_limit_per_byte * payload.length()
         gas = data_movement_gas + execution_gas
 
-        if has_guardian:
-            gas += self.config.extra_gas_limit_for_guarded_transactions
+        # add by default extra gas for guarded transactions
+        gas += DEFAULT_EXTRA_GAS_LIMIT_FOR_GUARDED_TRANSACTION
 
         return gas
 
-    def create_transaction(
+    def _create_transaction_intent(
             self,
             sender: IAddress,
             receiver: IAddress,
             data_parts: List[str],
-            execution_gas_limit: IGasLimit,
-            gas_limit_hint: Optional[IGasLimit],
-            gas_price: Optional[IGasPrice],
-            nonce: Optional[INonce],
-            value: Optional[ITransactionValue],
-            guardian: Optional[IAddress]) -> Transaction:
+            execution_gas_limit: int,
+            value: Optional[Union[str, int]] = None) -> TransactionIntent:
         data = self._build_transaction_payload(data_parts)
-        version = TRANSACTION_VERSION_DEFAULT
-        options = TRANSACTION_OPTIONS_DEFAULT
+        gas_limit = self._compute_gas_limit(data, execution_gas_limit)
 
-        has_guardian = True if guardian else False
-        gas_limit = gas_limit_hint or self._compute_gas_limit(data, execution_gas_limit, has_guardian)
+        transaction_intent = TransactionIntent()
 
-        return Transaction(
-            chain_id=self.config.chain_id,
-            sender=sender,
-            receiver=receiver,
-            gas_limit=gas_limit,
-            gas_price=gas_price,
-            value=value,
-            nonce=nonce,
-            data=data,
-            version=version,
-            options=options,
-            guardian=guardian
-        )
+        transaction_intent.sender = sender.bech32()
+        transaction_intent.receiver = receiver.bech32()
+        transaction_intent.gas_limit = gas_limit
+        transaction_intent.data = str(data).encode()
+        transaction_intent.value = str(value) if value else "0"
```

Comment:
We can have this as `int`, as well - even if the specs say that "intent.value" is a "string". It's a customization that works fine for Python.

Reply:
It is now of type `int`